micromagneticdata/combined_drive.py

```diff
@@ -0,0 +1,166 @@
+import functools
+import operator
+
+import ubermagutil as uu
+
+import micromagneticdata as md
+
+
+@uu.inherit_docs
+# @ts.typesystem(
+#     name=ts.Typed(expected_type=str),
+#     number=ts.Scalar(expected_type=int, unsigned=True),
+#     dirname=ts.Typed(expected_type=str),
+# )
+class CombinedDrive(md.AbstractDrive):
+    """Drive class.
+
+    This class provides utility for the analysis of individual drives.
+
+    Parameters
+    ----------
+    drives : List[Drive]
+
+    Raises
+    ------
+    ValueError
+
+        If the drives have different independent variables or less than two drives are
+        passed.
+
+    TypeError
+
+        If the passed objects are not of type ``micromagneticdata.Drive``.
+
+    Examples
+    --------
+    1. Getting drive object.
+
+    >>> import os
+    >>> import micromagneticdata as md
+    ...
+    >>> dirname = dirname=os.path.join(os.path.dirname(__file__),
+    ...                                'tests', 'test_sample')
+    >>> drive = md.Drive(name='system_name', number=0, dirname=dirname)
+
+    """
+
+    def __init__(self, *drives):
+        if len(drives) < 2:
+            raise ValueError("At least two drives must be pased.")
+        for drive in drives:
+            if not isinstance(drive, md.Drive):
+                raise TypeError(f"Object {drive}, {type(drive)=} is not of type Drive.")
```

Comment:
I am wondering whether we should forbid combining Min and TimeDrive, also what if the drives are not successive `drive0[-1] != drive1[0]`? Probably not... users should be clever enough.

Reply:
Combining Min and TimeDrive is implicitly forbidden through `ubermagtable`.

I wouldn't check for successive runs. I think this imposes too much restrictions. Maybe there are edge cases where other operations would make sense.

Reply:
That's a very good point :) I forgot about Table restrictions.